public/jsx/views/login.jsx

```diff
@@ -29,9 +29,17 @@ module.exports = React.createClass({
       method: 'post',
       url: '/api/auth/sign-in/',
       context: this,
-    }).then(function() {
+    }).then(function(data) {
       if (this.isMounted()) {
         this.setState({'is_logged_in': true}); // eslint-disable-line
+
+        console.log('setting CSRF token for subsequent requests:', data.csrf_token);
+        $.ajaxSetup({
```

Comment:
yeah, it looks like something you can run multiple times and not override old values. I'll test it and see. It's treating jQuery like a static store of saved data. If I were to do it some other way we'd still need the same concept of keeping a static store of data in memory.